### PR TITLE
Fix validation on multi-gpu, use less workers for dataloader, add slurm launche.

### PR DESCRIPTION
**Pull request by patil-suraj**

### CODE REVIEW

File: train_instruct_pix2pix_sdxl.py [hunk 1]
```diff
@@ -431,6 +434,83 @@ def parse_args():
 
     return args
 
+def log_validation(vae, unet, text_encoder_1, text_encoder_2, tokenizer_1, tokenizer_2, args, accelerator, weight_dtype, global_step):
```
Comment (patil-suraj) [inline train_instruct_pix2pix_sdxl.py:437]: factored it out in a function to make training loop clean.

File: train_instruct_pix2pix_sdxl.py [hunk 2]
```diff
@@ -1001,8 +1074,8 @@ def compute_time_ids(batch_size):
                 progress_bar.update(1)
                 global_step += 1
 
-                if global_step % args.checkpointing_steps == 0:
-                    if accelerator.is_main_process:
+                if accelerator.is_main_process:
```
Comment (patil-suraj) [inline train_instruct_pix2pix_sdxl.py:1077]: checkpointing and validation should only be done on main process

File: run.slurm [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+PROGRAM="train_instruct_pix2pix_sdxl.py \
+    --pretrained_model_name_or_path=$MODEL_ID \
+    --pretrained_vae_model_name_or_path=madebyollin/sdxl-vae-fp16-fix \
+    --dataset_path='pipe:aws s3 cp s3://muse-datasets/instructpix2pix-clip-filtered-wds/{000000..000062}.tar -' \
+    --use_ema \
+    --enable_xformers_memory_efficient_attention \
+    --resolution=256 --random_flip \
+    --per_gpu_batch_size=16 --gradient_accumulation_steps=4 \
+    --num_workers=4 \
```
Comment (patil-suraj) [inline run.slurm:64]: num_workers=48 is too much. That's 48 per GPU, so 384 in total. With those many workers, we'll send a lot of simultaneous requests to `s3`, which will block it, which results in broken pipe error.

**Review by sayakpaul — APPROVED** [review]
Thanks a lot!

File: train_instruct_pix2pix_sdxl.py [hunk 4]
```diff
@@ -868,24 +950,15 @@ def compute_time_ids(batch_size):
...
-            # Skip steps until we reach the resumed step
-            if (
-                args.resume_from_checkpoint
-                and epoch == first_epoch
-                and step < resume_step
-            ):
-                if step % args.gradient_accumulation_steps == 0:
-                    progress_bar.update(1)
-                continue
-
```
Comment (patil-suraj) [inline train_instruct_pix2pix_sdxl.py:882]: This is not required with `wds`, as with web dataset we do random resuming, which with large datasets performs similar to exact resuming. cf https://github.com/webdataset/webdataset/issues/177#issuecomment-1118186022

**Pull request merged**